### PR TITLE
Qualification tool: Detect JDBCRelation in eventlog

### PR DESCRIPTION
**Pull request by nartal1**
This fixes https://github.com/NVIDIA/spark-rapids/issues/3432 . It scans for JDBCRelation in eventlog and marks as potential problem. Added test to verify.

### CODE REVIEW

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala [hunk 1]
```diff
@@ -409,6 +409,11 @@ class QualificationSuite extends FunSuite with BeforeAndAfterEach with Logging {
     runQualificationTest(logFiles, "decimal_part_expectation.csv")
   }
 
+  test("test jdbc problematic") {
+    val logFiles = Array(s"$logDir/jdbc_eventlog.zstd")
```
[the remark this answers is not in the document]
Comment (nartal1) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala:413]: @tgravescs - I agree that we should have less hardcoded files but in this case I think hardcoded eventlog makes sense. Creating an eventlog to include `JDBCRelation` would involve reading a table from database(like mysql etc) along with Spark options to authenticate which seems overkill for testing.  Maybe we could remove `jdbc_expectation.csv` and instead assert for the fields which we are interested in. 
Please let me know your thoughts on this.

[the remark this answers is not in the document]
Comment (tgravescs) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala:413]: I'm fine with this one not changing, just note for future if we can easily do it in code lets do it that way.

**Comment by nartal1** [comment]
build

**Comment by nartal1** [comment]
@tgravescs PTAL

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala [hunk 2]
```diff
@@ -227,6 +227,7 @@ class QualificationAppInfo(
 
   private[qualification] def processSQLPlan(sqlID: Long, planInfo: SparkPlanInfo): Unit = {
     checkMetadataForReadSchema(sqlID, planInfo)
+    checkJdbcScan(sqlID, planInfo)
```
Comment (tgravescs) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala:230]: seems like we could easily have this apply to profiling tool but adding this same line into the ApplicationInfo class and updating tests and docs.

**Comment by nartal1** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**